src/lightbox/downloadFileToCache.js

```diff
@@ -0,0 +1,21 @@
+/* @flow strict-local */
+import RNFetchBlob from 'rn-fetch-blob';
+import { getMimeTypeFromFileExtension } from '../utils/url';
+
+/**
+ * Download a remote file to the app's cache directory.
+ *
+ * @param tempUrl A URL to the file. Should be a valid temporary URL generated
+ *     using `getFileTemporaryUrl`.
+ * @param fileName Name of the file to be downloaded. Should include the
+ *     extension.
+ */
+
+export default async (tempUrl: string, fileName: string): Promise<mixed> =>
+  RNFetchBlob.config({
+    path: `${RNFetchBlob.fs.dirs.CacheDir}/${fileName}`,
+    fileCache: true,
+    useDownloadManager: true,
+    mime: getMimeTypeFromFileExtension(fileName.split('.').pop()),
+    title: fileName,
+  }).fetch('GET', tempUrl);
```

Comment:
This is quite similar to the code in the Android case of `downloadImage`. That's fine -- it's more important to have it working correctly than to share code, and when I look closely I see that it actually wouldn't be simple to have the two versions share much code in a useful way -- but let's get the two versions right next to each other in the code. That way it's easy to compare them, and also when we're changing one it'll be easy to notice the other and consider whether we should be changing it the same way.

I think the best way to do that probably looks like:
* First, have a prep commit that renames `downloadImage.js` to like `download.js`, and turns the existing default-export into an export named `downloadImage`.
* Then, the commit that adds this version can just add another function to that file, with a name like `downloadFileToCache`.